src/NuGet.Clients/NuGet.VisualStudio.Contracts/NuGetInstalledPackage.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.VisualStudio.Contracts
+{
+    /// <summary>Information about an installed package</summary>
+    public sealed class NuGetInstalledPackage
+    {
+        /// <summary>The package id.</summary>
+        public string Id { get; }
+
+        /// <summary>The project's requested package range for the package.</summary>
+        /// <remarks>
+        /// If the project uses packages.config, this will be same as the installed package version.
+        /// If the project uses PackageReference, this is the version string in the project file, which may not match the resolved package version, and may not be single version string.
+        /// </remarks>
+        public string RequestedRange { get; }
+
+        /// <summary>The lowest version allowed by the <see cref="RequestedRange"/></summary>
+        public string RequestedVersion { get; }
```

Comment:
Think this should be "version' and the resolved one. 
If the old API has a bug, then this one has a bug.

Reply:
This would have the problem that @rrelyea was concerned about during our spec review where we return one value now, API users depend on certain behaviour, and when we fix the bug and start returning the actual installed version, their code misbehaves.

For what it's worth, I'm 100% in agreement. I think we should only have the "requested range" and "installed version", and consider this initial implementation as buggy returning the wrong version, whcih needs to be fixed.

Returning only `RequestedRange` and adding (installed) `Version` later would mean that the API would be unusable by Roslyn, because `IVsPackageInstaller` does not allow API callers to pass a `VersionRange`, only a `NuGetVersion` (or string that can be parsed into one). 

Therefore, to unblock the UI delay fix, we need this first implementation of the API to return a string parsable as a `NuGetVersion`. Do we return something that today returns the requested range's lower bound, and later changes behaviour to return the resolved version?  Or do we introduce a new property in a future update?

I prefer changing behaviour. But my memory of the spec review was concerns of changing behaviour was greater. Although most team members remained silent during that part of the discussion.

Reply:
> This would have the problem that @rrelyea was concerned about during our spec review where we return one value now, API users depend on certain behaviour, and when we fix the bug and start returning the actual installed version, their code misbehaves.

It's incorrect right now and no one has noticed it. If they do get the wrong version, with no package path, no one has complained about it failing today. 
I can't see how fixing the actual version doesn't make things better. 
I think we are being too risk averse in this case. We are fixing a bug.

Reply:
> Returning only RequestedRange and adding (installed) Version later would mean that the API would be unusable by Roslyn, because IVsPackageInstaller does not allow API callers to pass a VersionRange, only a NuGetVersion (or string that can be parsed into one).

I also think Roslyn can do one thing in 16.7, another thing in 16.8. 
It just makes me uneasy leaking "bug details" in the API design like this. 
In 16.8 we can provide an API that can accept a version range too if that's necessary.

Reply:
works for me (and I definately think that our install API should accept a version range, although we have to design how the API will design for packages.config projects). I'll change the PR in an update.